homeassistant/components/google_mail/__init__.py

```diff
@@ -41,13 +37,6 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     auth = AsyncConfigEntryAuth(session)
     try:
         await auth.check_and_refresh_token()
-    except HomeAssistantError as err:
-        resp = err.args[0]
-        if isinstance(resp, ClientResponseError) and 400 <= resp.status < 500:
-            raise ConfigEntryAuthFailed(
-                "OAuth session is not valid, reauth required"
-            ) from err
-        raise ConfigEntryNotReady from err
     except ClientError as err:
```

Comment:
We're still handling ClientError here. Decide what layer should handle the client errors. Don't mix it up.